eq-author-api/src/validation/customKeywords/validatePipingInTitle.js

```diff
@@ -0,0 +1,96 @@
+const {
+  PIPING_TITLE_DELETED,
+  PIPING_TITLE_MOVED,
+} = require("../../../constants/validationErrorCodes");
+const cheerio = require("cheerio");
+const getPreviousAnswersForPage = require("../../../src/businessLogic/getPreviousAnswersForPage");
+const { flatMap, compact } = require("lodash/fp");
+
+module.exports = function(ajv) {
+  ajv.addKeyword("validatePipingInTitle", {
+    $data: true,
+    validate: function isValid(
+      otherFields,
+      entityData,
+      fieldValue,
+      dataPath,
+      parentData,
+      fieldName,
+      questionnaire
+    ) {
+      isValid.errors = [];
+
+      const splitDataPath = dataPath.split("/");
+      const currentPage =
+        questionnaire.sections[splitDataPath[2]].pages[splitDataPath[4]];
+
+      const allPagesForQuestionnaire = flatMap(
+        section => section.pages,
+        questionnaire.sections
+      );
+      const allAnswersForQuestionnaire = compact(
+        flatMap(page => page.answers, allPagesForQuestionnaire)
+      );
+
+      const previousAnswersForPage = getPreviousAnswersForPage(
+        questionnaire,
+        currentPage.id,
+        true
+      );
+
+      const $ = cheerio.load(entityData);
+      const pipedIdList = [];
+
+      $("p")
```

Comment:
I'm not familiar with Cheerio. Our of curiosity, how come you have "p" in here? Is it to find paragraph tags or something?

Reply:
yes - all the sans that hold each individual piped in Answer or Metdata are contained in a single <p></P>